woocommerce/woocommerce-android
# Hide cards based on analytics configuration

Closes: #10702 

### Why
As part of Expanded Analytics Hub milestone 2, we plan to add support for customizing the cards we display on the Analytic Hub screen.

### Description
This PR supports displaying/hiding cards based on the card's configuration. It does that by observing the card configuration and updating the `AnalyticsViewState` to group all card states in a single field (cards). In Addition to the main changes, I also introduced a small refactor for changing the `cardId` from cards configuration to the actual `card` enum and created a new DataStore configuration for Analytics configuration.

The card order will be handled in a different PR

### Testing instructions
TC1
1. Open the app
2. Tap on see more
3. Tap on the pencil at the top of the Analytics Hub screen
4. Modify the card configuration
5. Tap on Save
6. Check that only the cards marked as selected are displayed in the Analytic Hub screen
7. Refresh the screen
8. Check that the expected cards are displayed

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/bcd9da4a-42a7-4128-ad73-aeacba46f2c2
